app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java

```diff
@@ -0,0 +1,212 @@
+        adb.setPositiveButton("Cancel request", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                activity.finish();
+                Intent intent = new Intent(MakeRequestActivity.this, MainActivity.class);
+                startActivity(intent);
+            }
+        });
+        adb.setNegativeButton("Cancel", null );
+        adb.show();
+    }
+
+    /**
+     * Use the FareCalculator to estimate the fare between the user-selected start and end locations.
+     * The start and end locations must have both been selected before the fare can be estimated.
+     */
+    public void estimateFare() {
+        if(start == null || end == null) {
+            Toast.makeText(activity, "You must first select a start and end location", Toast.LENGTH_SHORT).show();
+        } else {
+            // TODO use FareCalculator once available
+            // the MockFareCalculator generates random numbers so we can see different values on the display
+            MockFareCalculator fc = new MockFareCalculator(start, end);
+            int fareEstimate = fc.getEstimate();
+
+            TextView fareTextView = (TextView) findViewById(R.id.textView_fareEstimate);
+            fareTextView.setText(formatFare(fareEstimate));
+
+            fareEstimated = fareEstimate;
+        }
+    }
+
+    /**
+     * Format the fare we get as a double as a string to be printed on the screen.
+     * This string will be preceded by a dollar sign (are we concerned about locale?)
+     * and will be to 2 decimal places.
+     * @param intFare
+     * @return String
+     */
+    // TODO deprecate once toString is available in FareCalculator
+    public String formatFare(int intFare) {
+        double fare = ((double) intFare)/100;
+        String str = String.format(Locale.getDefault(),"%.0f",fare/1) + ".";
+        String dec = String.format(Locale.getDefault(),"0%.0f",(fare%1)*100);
+        // format the fare as a string with 2 decimal points
+        str +=  dec.substring(dec.length()-2, dec.length());
+        return str;
+    }
+
+    /**
+     * When the submit button is pressed
+     *      Create a Request and add it to the request controller
+     *      Save the request (elasticsearch...through request controller?)
+     *      Return to MainActivity
+     */
+    public void submitRequest() {
+        RequestController rc = new RequestController();
+        UserController uc = new UserController();
+
+        if(start == null || end == null) {
+            Toast.makeText(activity, "You must first select a start and end location", Toast.LENGTH_SHORT).show();
+        } else if (fareEstimated != -1) {
+            Toast.makeText(activity, "You must first estimate the ride fare", Toast.LENGTH_SHORT).show();
+        } else {
+            User user = uc.getLoggedInUser();
```

Comment:
These checks should be done within Request Controller. 
That ensures a singular area for checks (if we ever have more than one area to create a request) :)
It also makes the view not responsible for things other than displaying and informing the controller of a potential request. 
Check out how I handle this sort of thing in the new user activity :)